apps/keyboard/js/views/swiping_detector.js

```diff
@@ -0,0 +1,152 @@
+'use strict';
+
+(function(exports) {
+
+function SwipingDetector(element) {
+  this.element = element;
+  this.state = 'init';
```

Comment:
`this.state = this.STATE_INIT` and
`SwipingDetector.prototype.STATE_INIT = 0`

is the better approach for a enum value.